### PR TITLE
Allow INFN redirector to be used when no sites are found with -ws option

### PR DESCRIPTION
**Pull request by andreypz**
This introduces  a new feature in `build_datasets` script, which allows to use INFN redirector prefix in cases when no white-listed sites were found f or a given file (using the `-ws` option). At the moment a Warning is printed when this happens, which may be suppressed in the future.

My use case is the following. Most of the datasets are copied to some German T2s but not all of them. I want to preference those sites, but when the sample is not there, just use the redirector:
```
build_datasets --cfg samples.json -o -ws T2_DE_RWTH  --use-infn
```
The behavior should be the same as before, when the flag is not used.

### CODE REVIEW

File: pocket_coffea/utils/rucio.py [hunk 1]
```diff
@@ -191,11 +192,19 @@ def get_dataset_files_replicas(
                     )
                     outsite.append(site)
                     found = True
+                    
+            if not found and use_infn:
+                # The file was not found at any of the allowed sites
+                # But with this option we add the INFN redirector prefix
+                if len(list(rses.keys())) != 0:
```
Comment (valsdav) [inline pocket_coffea/utils/rucio.py:199]: we are missing an `else` here, raising an exception about a dataset with completely no sites available

**Review by valsdav — CHANGES_REQUESTED** [review]
Thanks! I agree on this option. Just some questions

File: pocket_coffea/scripts/dataset/build_datasets.py [hunk 2]
```diff
@@ -49,9 +49,14 @@
 @click.option(
     "-ws",
     "--allowlist-sites",
-    type=str,
     multiple=True,
 )
+@click.option(
+    "--use-infn",
```
[the remark this answers is not in the document]
Comment (andreypz) [reply pocket_coffea/scripts/dataset/build_datasets.py]: Ok, changed that to `--include-redirector`

File: pocket_coffea/scripts/dataset/build_datasets.py [hunk 3]
```diff
@@ -49,9 +49,14 @@
...
+    is_flag=True,
+    default=False,
+    help="In case -ws option is used but a file is not found on any of the allowed sites, the INFN redirector prefix will be used instead, when this option is specified."
```
[the remark this answers is not in the document]
Comment (andreypz) [reply pocket_coffea/scripts/dataset/build_datasets.py]: In fact, in my previous version it was only applied for `-ws` option.
With the latest commit t should also work for for `-rs` and `-bs`, so updated this as well

File: pocket_coffea/utils/rucio.py (hunk 1 above)
Comment (andreypz) [reply pocket_coffea/utils/rucio.py]: I think it was not necessary. If `found=False` at the end, the message would be displayed later.
Also in the updated version, if the `found` flag not set, the exception will rise at the end of the function

**Comment by valsdav** [comment]
Thanks for the changes, I think we are missing only adding the option in the documentation of the analysis example and `Dataset handling` pages. May you add it there please?

**Comment by andreypz** [comment]
Ok updated documentation as well. 
There may be some spurious changes due to removing white spaces at the end of lines, should be harmless

**Comment by valsdav** [comment]
Thanks!! ready to go!

**Pull request merged**